src/wp-includes/class-wp-theme-json.php

```diff
@@ -1609,46 +1635,41 @@ public function get_svg_filters( $origins ) {
 	}
 
 	/**
-	 * Returns whether a presets should be overridden or not.
+	 * For metadata values that can either be booleans or paths to booleans, gets the value.
 	 *
-	 * @since 5.9.0
+	 * ```php
+	 * $data = array(
+	 *   'color' => array(
+	 *     'defaultPalette' => true
+	 *   )
+	 * );
+	 *
+	 * self::get_metadata_boolean( $data, false );
+	 * // => false
 	 *
-	 * @param array      $theme_json The theme.json like structure to inspect.
-	 * @param array      $path Path to inspect.
-	 * @param bool|array $override Data to compute whether to override the preset.
+	 * self::get_metadata_boolean( $data, array( 'color', 'defaultPalette' ) );
+	 * // => true
+	 * ```
+	 *
+	 * @since 5.9.0
+	 * 
+	 * @param array      $data The data to inspect.
+	 * @param bool|array $path Boolean or path to a boolean.
 	 * @return boolean
 	 */
-	private static function should_override_preset( $theme_json, $path, $override ) {
-		if ( is_bool( $override ) ) {
-			return $override;
+	private static function get_metadata_boolean( $data, $path, $default = false ) {
```

Comment:
Is it possible to use old logic from `should_override_preset` instead of introducing a new method and removing the old one?

This method won't be private after https://github.com/WordPress/wordpress-develop/pull/2322.

Reply:
It is possible. I would strongly advise against using the old one.

The old method is confusing because it returns either the boolean or the _inverse_ of the boolean when a path is passed. Making the return consistent—the boolean or the boolean at the end of a path—is much easier to understand. And it doesn't require checking if the path argument is an array at both the call site and the method.